### PR TITLE
Handle custom manifest name

### PR DESCRIPTION
**Pull request by IanButterworth**
Paired with https://github.com/JuliaLang/julia/pull/44286

### CODE REVIEW

File: src/Types.jl
Comment (DilumAluthge) [inline src/Types.jl:187]:
```suggestion
            man_name = only(man_names)
            man_path = joinpath(env_path, man_name)
            if strict && !isfile(man_path)
                return nothing
            else
                return joinpath(env_path, man_name)
```

File: src/Types.jl
Comment (DilumAluthge) [inline src/Types.jl:181]:
```suggestion
    @static if Base.VERSION >= v"1.9.0-"
        man_names = Base.manifest_names()
    else
        man_names = Base.manifest_names
    end
```

Comment (IanButterworth) [reply src/Types.jl:181]: This will have to follow. We can't merge this kind of catch before it's merged in Base

**Comment by IanButterworth** [comment]
Errm..
```
ERROR: LoadError: UndefVarError: man_names not defined
Stacktrace:
 [1] top-level scope
   @ ~/work/Pkg.jl/Pkg.jl/src/Types.jl:191
```

Is it me, or does this look like the purity PR deleting `man_names`.. I can't see how it's not defined on 191

**Comment by DilumAluthge** [comment]
You could try building Julia from source with the commit right before the purity modeling PR was merged.

**Comment by DilumAluthge** [comment]
I think you might have a syntax error somewhere.

**Comment by DilumAluthge** [comment]
You have one too many `end` in that function.

**Comment by DilumAluthge** [comment]
You have an `end` on 188, 189, and 190. One of those shouldn't be there.

File: src/Types.jl
Comment (DilumAluthge) [reply src/Types.jl:181]: Just as a short-term stop-gap, how about this?

```suggestion
    @static if Base.manifest_names isa Tuple
        man_names = Base.manifest_names()
    else
        man_names = Base.manifest_names
    end
```

Then we can replace it with the `Base.VERSION` check once the Base PR is merged and the nightly binaries catch up.

Comment (DilumAluthge) [reply src/Types.jl:181]: Or, if you want to keep the ternary:

```suggestion
    man_names = @static Base.manifest_names isa Tuple ? Base.manifest_names : Base.manifest_names()
```

**Review by DilumAluthge — APPROVED** [review]
I'd like that `@static` to be added, but other than that, this looks good to me.

**Comment by DilumAluthge** [comment]
Once this PR is merged, we can bump the Pkg version on Julia master, and after that bump PR is merged, we can rebase https://github.com/JuliaLang/julia/pull/44286.

File: src/Types.jl
Comment (DilumAluthge) [inline src/Types.jl:189]: @IanButterworth I may be missing something, but now that we guarantee that `man_names` is always the same length as `Base.project_names`, can we not just delete lines 182 to 189?

Comment (IanButterworth) [reply src/Types.jl:189]: Sure but it will check for the file twice if it doesn't exist.

Comment (DilumAluthge) [reply src/Types.jl:189]: Ah. Yeah, it's probably good to avoid doing unnecessary filesystem operations.

File: src/Types.jl
Comment (DilumAluthge) [inline src/Types.jl:183]: We can get rid of one of these calls to `first(man_names)`.

```suggestion
    first_man_name = first(man_names)
    if first_man_name == last(man_names) # indicative of a custom manifest name
        man_path = joinpath(env_path, first_man_name)
```

**Pull request merged**
